code/modules/pai/shell.dm

```diff
@@ -1,155 +1,174 @@
-			return FALSE
-	forceMove(get_turf(card))
-	card.forceMove(src)
-	if(client)
-		client.perspective = EYE_PERSPECTIVE
-		client.eye = src
-	set_light(0)
-	icon_state = "[chassis]"
-	held_state = "[chassis]"
-	visible_message(span_boldnotice("[src] folds out its holochassis emitter and forms a holoshell around itself!"))
-	holoform = TRUE
+/**
+ * Checks if we are allowed to interact with a radial menu
+ *
+ * @param {atom} anchor - The atom that is anchoring the menu.
+ *
+ * @returns {boolean} - TRUE if we are allowed to interact with the menu,
+ * 	FALSE otherwise.
+ */
+/mob/living/silicon/pai/proc/check_menu(atom/anchor)
+	if(incapacitated())
+		return FALSE
+	if(get_turf(src) != get_turf(anchor))
+		return FALSE
+	if(!isturf(loc) && loc != card)
+		balloon_alert(src, "can't do that here")
+		return FALSE
+	return TRUE
 
-/mob/living/silicon/pai/proc/emittercool()
-	emittersemicd = FALSE
+/**
+ * Sets a new holochassis skin based on a pAI's choice.
+ *
+ * @returns {boolean} - True if the skin was successfully set.
+ * 	FALSE otherwise.
+ */
+/mob/living/silicon/pai/proc/choose_chassis()
+	var/list/skins = list()
+	for(var/holochassis_option in possible_chassis)
+		var/image/item_image = image(icon = src.icon, icon_state = holochassis_option)
+		skins += list("[holochassis_option]" = item_image)
+	sort_list(skins)
+	var/atom/anchor = get_atom_on_turf(src)
+	var/choice = show_radial_menu(src, anchor, skins, custom_check = CALLBACK(src, .proc/check_menu, anchor), radius = 40, require_near = TRUE)
+	if(!choice)
+		return FALSE
+	set_holochassis(choice)
+	balloon_alert(src, "[choice] composite engaged")
+	update_resting()
+	return TRUE
 
+/**
+ * Returns the pAI to card mode.
+ *
+ * @param {boolean} force - If TRUE, the pAI will be forced to card mode.
+ *
+ * @returns {boolean} - TRUE if the pAI was forced to card mode.
+ * 	FALSE otherwise.
+ */
 /mob/living/silicon/pai/proc/fold_in(force = FALSE)
-	emittersemicd = TRUE
+	holochassis_ready = FALSE
 	if(!force)
-		addtimer(CALLBACK(src, .proc/emittercool), emittercd)
+		addtimer(VARSET_CALLBACK(src, holochassis_ready, TRUE), HOLOCHASSIS_COOLDOWN)
 	else
-		addtimer(CALLBACK(src, .proc/emittercool), emitteroverloadcd)
+		addtimer(VARSET_CALLBACK(src, holochassis_ready, TRUE), HOLOCHASSIS_OVERLOAD_COOLDOWN)
 	icon_state = "[chassis]"
 	if(!holoform)
 		. = fold_out(force)
-		return
+		return FALSE
 	visible_message(span_notice("[src] deactivates its holochassis emitter and folds back into a compact card!"))
 	stop_pulling()
 	if(istype(loc, /obj/item/clothing/head/mob_holder))
-		var/obj/item/clothing/head/mob_holder/MH = loc
-		MH.release(display_messages = FALSE)
+		var/obj/item/clothing/head/mob_holder/mob_head = loc
+		mob_head.release(display_messages = FALSE)
 	if(client)
 		client.perspective = EYE_PERSPECTIVE
 		client.eye = card
-	var/turf/T = drop_location()
-	card.forceMove(T)
+	var/turf/turf = drop_location()
```

Comment:
nitpicky buuuut you should avoid just using `turf` as a replacement for `T` cus its nearly as bad